Src/library/net/network/socket/network_socket.e

```diff
@@ -163,25 +163,57 @@ feature -- Status report
 		local
 			retval: INTEGER
 		do
-			retval := c_check_exception_with_timeout (descriptor, timeout)
+			retval := c_check_exception_with_timeout (descriptor, timeout_ns)
 			Result := (retval > 0)
 		end
 
 	timeout: INTEGER
 			-- Duration of timeout in seconds
+		obsolete
+			"Use `timeout_ns` using nanoseconds [2018-01-15]"
+		do
+			Result := (timeout_ns // one_second_in_nanoseconds).to_integer_32
+			if Result = 0 then
+					-- 0 may have different meaning, so return the lowest possible value.
+				Result := 1
+			end
+		end
+
+	timeout_ns: NATURAL_64
+			-- Duration of timeout in nanoseconds.
 
 	recv_timeout: INTEGER
 			-- Receive timeout in seconds on Current socket.
+		obsolete
+			"Use `recv_timeout_ns` using nanoseconds [2018-01-15]"
+		do
+			Result := (recv_timeout_ns \\ one_second_in_nanoseconds).to_integer_32
```

Comment:
Should it be '//'?
Also, see a comment about possible overflow that results in a negative result.

Reply:
fixed